### ISSUE TITLE
Hint text semantics to be excluded in a11y read out if textfield in not empty and label text is provided 

### ISSUE THREAD

**Issue by stellaseahll** [flutter/flutter#114511]
Internal: b/258261921

Currently, hint text can either be used to (1) provide hint text for users, or (2) as a label for the text field. 
E.g. to enter phone number, hint text can be either (1) 0000 - 0000 or (2) Phone number. 

For a11y users, the hint text is always read regardless of use case (1) or (2) and this is problematic for use case (1).

E.g. when the text field is filled with 1234 - 5678, it reads both the text and hint text "1234 - 5678, 0000 - 0000"

Can we tweak the current feature to exclude the hint from a11y if the textfield is not empty AND a dedicated label text is given?

**Comment by goderbauer** [comment]
Internal bug for this is b/255727805

/cc @justinmc @chunhtai 

> Can we tweak the current feature to exclude the hint from a11y if the textfield is not empty AND a dedicated label text is given?

This sounds reasonable to me as long as the hint is included if there is no explicit label given because some app designs use the hint as a label in that case and we don't want to loose that information there.

**Comment by chunhtai** [comment]
This sounds reasonable we should exclude the hintext if there is text in the textfield.

**Comment by goderbauer** [comment]
> 
This sounds reasonable we should exclude the hintext if there is text in the textfield.

Only if there is a separate label text, though. Otherwise, the hint is likely the label.

**Comment by chunhtai** [comment]
Should be a one line fix
https://github.com/flutter/flutter/blob/78dbe6661b3557c2355cad401287d0bce3479c06/packages/flutter/lib/src/material/input_decorator.dart#L2148

@hangyujin can you take a look at this?
